API/models/accounts.go

```diff
@@ -65,79 +106,206 @@ func (account *Account) Create() (map[string]interface{}, string) {
 	account.Password = string(hashedPassword)
 
 	ctx, cancel := u.Connect()
-	search := GetDB().Collection("account").FindOne(ctx, bson.M{"email": account.Email})
-	if search.Err() != nil {
-		GetDB().Collection("account").InsertOne(ctx, account)
-	} else {
+	res, err := GetDB().Collection("account").InsertOne(ctx, account)
+	if err != nil {
 		return u.Message(false,
-			"Error: User already exists:"), "clientError"
+			"DB error when creating user: "+err.Error()), "internal"
+	} else {
+		account.ID = res.InsertedID.(primitive.ObjectID)
 	}
-
 	defer cancel()
 
 	//Create new JWT token for the newly created account
-	tk := &Token{Email: account.Email}
-	token := jwt.NewWithClaims(jwt.GetSigningMethod("HS256"), tk)
-	tokenString, _ := token.SignedString([]byte(os.Getenv("token_password")))
-
-	account.Token = tokenString
-
+	account.Token = GenerateToken(account.Email, account.ID, token_expiration)
 	account.Password = ""
 
 	response := u.Message(true, "Account has been created")
 	response["account"] = account
 	return response, ""
 }
 
+func validatePasswordFormat(password string) string {
+	if len(password) < 7 {
+		return "Please provide a password with a length greater than 6"
+	}
+	return ""
+}
+
+func comparePasswordToAccount(account Account, inputPassword string) (string, string) {
+	println(inputPassword)
+	println(account.Password)
```

Comment:
printing passwords on the standard output seems a bit dangerous to me

Reply:
that shouldn't be there anymore, ops